include/drivers/console/bm_serial.h

```diff
@@ -40,11 +40,18 @@ enum BM_PAYLOAD_TYPE
 	BM_IEEE802154   = 1,
 };
 
-typedef struct bm_man_rx_t
+typedef struct bm_test_ret_t
+{
+    int         retval;
+    uint16_t    length;
+    uint8_t*    buf_ptr;
+} bm_test_ret_t;
```

Comment:
nit: if `bm_serial_process_byte` will be called outside of the unit test, I'd remove the _test_ part of the name. Just to make it clear it's just the normal return value from that function.